fenwick/src/main/java/org/opencadc/fenwick/InventoryHarvester.java

```diff
@@ -185,43 +194,108 @@ public InventoryHarvester(Map<String, Object> daoConfig, URI resourceID, Artifac
      */
     @Override
     public void run() {
+        boolean retry;
+        int retryCount = 1;
+        int sleepSeconds = INITIAL_SLEEP_TIMEOUT;
+
         while (true) {
             try {
                 final Subject subject = SSLUtil.createSubject(new File(CERTIFICATE_FILE_LOCATION));
-                Subject.doAs(subject, (PrivilegedExceptionAction<Void>) () -> {
-                    doit();
-                    return null;
+                int retries = retryCount;
+                int timeout = sleepSeconds;
+
+                retry = Subject.doAs(subject, (PrivilegedAction<Boolean>) () -> {
+                    boolean isRetry = true;
+                    try {
+                        doit();
+                        isRetry = false;
+                        // catch exceptions resulting in a retry
+                    } catch (ResourceNotFoundException ex) {
```

Comment:
Why would you create new error messages with the http status code (which will eventually be wrong as code elsewhere evolves) when you could just do a multi-catch of all non-fatal exceptions and use the exception name in the error message. Even ex.toString() would be more or less OK.

Reply:
Used a multi-catch for the run() exceptions that result in a retry, and use the exception message for the retry logging.